src/android/CameraLauncher.java

```diff
@@ -494,16 +494,18 @@ private void processResultFromCamera(int destType, Intent intent) throws IOExcep
         // in the gallery and the modified image is saved in the temporary
         // directory
         if (this.saveToPhotoAlbum) {
-            galleryUri = Uri.fromFile(new File(getPicturesPath()));
+            GalleryPathVO galleryPathVO = getPicturesPath();
+            galleryUri = Uri.fromFile(new File(galleryPathVO.getGalleryPath()));
 
             if (this.allowEdit && this.croppedUri != null) {
                 writeUncompressedImage(croppedUri, galleryUri);
             } else {
-                Uri imageUri = this.imageUri;
-                writeUncompressedImage(imageUri, galleryUri);
+                if (Build.VERSION.SDK_INT <= Build.VERSION_CODES.P) { // Between LOLLIPOP_MR1 and P
```

Comment:
I kinda forget android behaviour, can we double check that `Build.VERSION_CODES.P` is safe to use on devices that are API  < 28?

Ie: Android 5.1 simulator, with an app built with target API 29.

Reply:
Yes, this works.
- I dit test with target API 29, and 5.1 emulator. It works.

The Build.VERSION_CODES are compile-time constants that gets inlined in the generated bytecode. Because of that, the symbol does not need to be resolved at runtime.

Some additional context:
- I did also make the decision to use smaller or equal to Build.VERSION_CODES.P instead of smaller than Build.VERSION_CODES.Q . Because When you downscale the compile version to 28, it keeps compiling and working. 
- But you cannot compile with version 27. Becuase VERSION_CODES.P is not available in that sdk.

![image](https://user-images.githubusercontent.com/33040889/95965581-6280c480-0e0a-11eb-8582-ff70d190dd56.png)

Reply:
Ok, even though this normally would be the acceptable approach because Google Play enforces apps to be built with target SDK 28, (and soon to be forced for all apps to use API target 29)

Cordova is suppose to support enterprise environments where the play store is not involved in their process, and therefore they may use a lower target SDK than what Google demands. This means we can't actually use the `VERSION_CODES` constants unless if they are introduced in/before our min sdk level. So I guess for now the solution to this is to change `Build.VERSION_CODES.P` to a hard-code numerical `28`.

This isn't documented in the Cordova docs, I'm requesting the actual minimum target API we support to be defined so that this can actually be documented.

Reply:
@breautek 

I digged a little deeper.
When I change the Build.VERSION_CODES_P to the numerical 28 , I am able to compile and run it with the defaultCompileSdkVersion starting from 22 and higher.
So I changed the code to the 28 number.

![image](https://user-images.githubusercontent.com/33040889/96040965-6690fe80-0e6b-11eb-9327-36a314397fe1.png)

But is it really necessary to be able to compile with the sdk version equal to 22?

Kind regards
Pieter

Reply:
> But is it really necessary to be able to compile with the sdk version equal to 22?

My personal opinion is no.... but I think that's what our min requirement is going to be. I'm about to write a new thread in the dev mail to see if the community can come to a consensus and so that we can document what Cordova supports for enterprise users. I'm assuming the consensus will be "whatever what the min sdk version is", which is currently 22.

Obviously this only applies to people who aren't publishing to the google play store.